src/Phinx/Console/Command/Breakpoint.php

```diff
@@ -62,22 +62,13 @@ protected function execute(InputInterface $input, OutputInterface $output)
     {
         $this->bootstrap($input, $output);
 
-        $environment = $input->getOption('environment');
         $version = (int)$input->getOption('target') ?: null;
         $removeAll = $input->getOption('remove-all');
         $set = $input->getOption('set');
         $unset = $input->getOption('unset');
 
+        $environment = $this->getEnvironment($input, $output);
```

Comment:
It doesn't look like `getEnvironment()` caches the result. Instead, it will trigger fallback logic and logging each time it is called.

Reply:
This is true of the `bootstrap` function as well. I guess the name here is a bit misleading on what it does, but I could not come up with a better one as I'm not sure I see the value in doing caching on this method.

Reply:
The value is a `get()` that doesn't logic fallback and failure logic each time. This code is trying to do too much at once.